### PR TITLE
James/ci

### PR DESCRIPTION
**Pull request by jameshod5**
Added same CI from fair-mast repo. I have included ruff fixes, mainly just unused variables. Tests that use the MAST client have been skipped.

- Check the changes made.

The CI should run with this PR and complete.

### CODE REVIEW

File: tests/test_reader.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def test_list_signals_exclude_raw():
     info = signals[0]
     assert info.name == "abm/calib_shot"
 
-
+@pytest.mark.skip(reason="Uses MAST Client, not available on CSD3")
```
Comment (samueljackson92) [inline tests/test_reader.py:37]: Don't think we should mentioned CSD3. Just say that the pyuda client is not available.

File: tests/test_writer.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import pytest
 import zarr
 import xarray as xr
-from src.archive.writer import DatasetWriter
+from src.writer import DatasetWriter
```
Comment (samueljackson92) [inline tests/test_writer.py:4]: Fixing my bugs. 👍

**Review by samueljackson92 — CHANGES_REQUESTED** [review]
Mostly looks good to me, but please change the reason strings to not mention CSD3.

File: tests/test_reader.py (hunk 1 above)
Comment (jameshod5) [reply tests/test_reader.py]: done :)

**Review by samueljackson92 — APPROVED** [review]
LGTM

**Pull request merged**